vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericSetVariableSteps.java

```diff
@@ -0,0 +1,149 @@
+     * <ul>
+     * <li>Saves the value of attribute with name <i>attributeName</i> to the <i>variableName</i>
+     * </ul>
+     * @param attributeName the name of the attribute (for ex. 'name', 'id')
+     * @param scopes The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of variable's scope<br>
+     * <i>Available scopes:</i>
+     * <ul>
+     * <li><b>STEP</b> - the variable will be available only within the step
+     * <li><b>SCENARIO</b> - the variable will be available only within the scenario
+     * <li><b>STORY</b> - the variable will be available within the whole story
+     * <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     * </ul>
+     * @param variableName A name under which the value should be saved
+     */
+    @When("I save `$attributeName` attribute value of context element to $scopes variable `$variableName`")
+    public void saveContextElementAttributeValueToVariable(String attributeName, Set<VariableScope> scopes,
+            String variableName)
+    {
+        saveVariableIfContextElementPresent(contextElement -> getAssertedAttributeValue(contextElement, attributeName),
+                scopes, variableName);
+    }
+
+    /**
+     * Extracts the value of <b>attribute</b> of element found by <b>locator</b> and saves it to the <b>variable</b>
+     * with the specified <b>variableName</b>
+     * Actions performed at this step:
+     * <ul>
+     * <li>Saves the value of attribute with name <i>attributeName</i> to the <i>variableName</i>
+     * </ul>
+     * @param attributeName the name of the attribute (for ex. 'name', 'id')
+     * @param locator locator to find an element
+     * @param scopes The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of variable's scope<br>
+     * <i>Available scopes:</i>
+     * <ul>
+     * <li><b>STEP</b> - the variable will be available only within the step
+     * <li><b>SCENARIO</b> - the variable will be available only within the scenario
+     * <li><b>STORY</b> - the variable will be available within the whole story
+     * <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     * </ul>
+     * @param variableName A name under which the value should be saved
+     */
+    @When("I save `$attributeName` attribute value of element located `$locator` to $scopes variable `$variableName`")
+    public void saveAttributeValueOfElementByLocatorToVariable(String attributeName, Locator locator,
+            Set<VariableScope> scopes, String variableName)
+    {
+        locator.getSearchParameters().setVisibility(Visibility.ALL);
+        Object value = baseValidations.assertElementExists("The element to extract the attribute", locator)
+                                      .map(element -> getAssertedAttributeValue(element, attributeName))
+                                      .orElse(null);
+        bddVariableContext.putVariable(scopes, variableName, value);
```

Comment:
the part of the code can be extracted to a new method, and then re-used:
```java
private void putVariable(Optional<WebElement> element, Function<WebElement, Object> variableMapper, Set<VariableScope> scopes, String variableName)
{
    Object value = element.map(variableMapper).orElse(null);
    bddVariableContext.putVariable(scopes, variableName, value);
}
```

Reply:
I will add this, but imho I doesn't look like a thing that makes the code more readable